ably/rest/rest.py

```diff
@@ -77,23 +79,23 @@ async def __aenter__(self):
         return self
 
     @catch_all
-    async def stats(self, direction=None, start=None, end=None, params=None,
-                    limit=None, paginated=None, unit=None, timeout=None):
+    async def stats(self, direction: Optional[str] = None, start=None, end=None, params: Optional[dict] = None,
+                    limit: Optional[int] = None, paginated=None, unit=None, timeout=None):
         """Returns the stats for this application"""
-        params = format_params(params, direction=direction, start=start, end=end, limit=limit, unit=unit)
-        url = '/stats' + params
+        formatted_params = format_params(params, direction=direction, start=start, end=end, limit=limit, unit=unit)
+        url = '/stats' + formatted_params
         return await PaginatedResult.paginated_query(
             self.http, url=url, response_processor=stats_response_processor)
 
     @catch_all
-    async def time(self, timeout=None):
+    async def time(self, timeout: Optional[float] = None):
```

Comment:
For this method, since `r.to_native()[0]` can return pretty much anything it's important to set the return type explicitly
```suggestion
    async def time(self, timeout: Optional[float] = None) -> float:
```